packages/block-editor/src/components/use-style/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { useBlockEditContext } from '../block-edit';
+import { store as blockEditorStore } from '../../store';
+import { getResolvedStyleVariable } from '../../utils/style-variable-resolution';
+
+/**
+ * Hook that retrieves the global styles of a block.
+ * It works with nested objects using by finding the value at path.
+ *
+ * @param {string|Array} path The path to the setting.
+ *
+ * @return {any} Returns the style value defined for the path.
+ *
+ * @example
+ * ```js
+ * const backgroundColor = useStyle( 'color.background' );
+ * ```
+ */
+export default function useStyle( path ) {
+	const { name: blockName } = useBlockEditContext();
+
+	const setting = useSelect(
+		( select ) => {
+			const settings = select( blockEditorStore ).getSettings();
+			const settingsForBlock = get( settings, [
```

Comment:
mmm, was looking at what sort of data this was using. Some braindump as to how / why inheritance would increase the complexity to take into account:

- Some CSS properties are inherited, some other don't.
- A block can be top-level but also can be nested within other blocks: to determine the value we'd need to traverse up the tree and consolidate all values of the chain.
- CSS custom properties: if the theme uses them, this breaks the CSS inheritance model for the property.

I agree this PR should be limited to only take data from the specific block.